src/DataMigration/DataMigration.Autorest/custom/Helpers/LoginMigrations.ps1

```diff
@@ -56,5 +56,196 @@ function  Get-DefaultLoginMigrationsOutputFolder {
+                    }
+                }
+            }
+
+            if ($LatestNugetOrgDetails.NameAndVersion -gt $LatestLocalNameAndVersion -and $userUpdateConsent -eq "yes")
+            {
+                #Update is available
+                $DownloadUrl = "https://www.nuget.org/api/v2/package/$PackageId/$($LatestNugetOrgDetails.Version)"
+
+                #Checking if LatestNugetFolder Path is valid or not
+                if(-Not (Test-Path $LatestNugetFolder.Value))
+                {
+                    $null = New-Item -Path $LatestNugetFolder.Value -ItemType "directory";
+                }
+
+                Write-Host "Downloading the latest Login migration console app nupkg: $($LatestNugetOrgDetails.NameAndVersion) ..."
+                Invoke-WebRequest $DownloadUrl -OutFile "$($LatestNugetFolder.Value)\\$($LatestNugetOrgDetails.NameAndVersion).nupkg"
+
+                $ToolsPathExists = Test-Path -Path (Join-Path -Path $LatestNugetFolder.Value -ChildPath "tools");
+
+                if ($ToolsPathExists -eq $False)
+                {
+                    $Nugets = Get-ChildItem -Path $LatestNugetFolder.Value -Filter "$PackageId.*.nupkg";
+
+                    if ($Nugets.Length -gt 0)
+                    {
+                        Write-Host "Extracting the latest Login migration console app nupkg: $($LatestNugetOrgDetails.NameAndVersion) ..."
+                        $Nugets = $Nugets | Sort-Object -Property Name -Descending;
+                        $LatestNugetPath = $Nugets[0].FullName;
+                        Expand-Archive -Path $LatestNugetPath -DestinationPath $LatestNugetFolder.Value;
+                    }
+                }
+
+                #Check if update was successful
+                $TestPathResult = Test-Path -Path "$($LatestNugetFolder.Value)\$ExePath"
+
+                $NugetVersions = Get-ChildItem -Path $DownloadsFolder -Filter "$PackageId.*";
+                $NugetVersions = $NugetVersions | Sort-Object -Property Name -Descending
+
+                if($TestPathResult)
+                {
+                    Write-Host "Removing all older Login migration console apps..."
+                    #Remove all other NuGet versions except for the version just downloaded
+                    for ($NugetIndex = 0; $NugetIndex -lt $NugetVersions.Length; $NugetIndex = $NugetIndex + 1)
+                    {
+                        if($NugetVersions[$NugetIndex].Name -ne $LatestNugetOrgDetails.NameAndVersion)
+                        {
+                            Remove-Item -Path $NugetVersions[$NugetIndex].FullName -Recurse -Force
+                        }
+                    }
+                }
+                else
+                {
+                    if($NugetVersions.Length -gt 0)
+                    {
+                        $LatestNugetFolder.Value = $NugetVersions[0].Name;
+                    }
+                }
+            }
+        }
+        catch
```

Comment:
nit: Lets not `try-catch` if we are just going to `throw`.

Reply:
Agree. Removed it. Thanks for pointing out.